src/App.php

```diff
@@ -123,6 +125,24 @@ protected function configureIoc( \Slim\Helper\Set $container ) {
 				$c->i18nCache, $c->settings['i18n.default'], $c->log
 			);
 		} );
+		// Set up cache (Redis if config is provided, otherwise local filesystem).
+		$container->singleton( 'cache', function( $c ) {
+			$cache = new Pool();
+			if ( Config::getStr( 'REDIS_HOST' ) ) {
+				$driver = new Redis( [
+					'servers' => [
+						'server_1' => [
+							'server' => Config::getStr( 'REDIS_HOST' ),
+							'port' => Config::getStr( 'REDIS_PORT' ),
+						],
+					]
+				] );
+			} else {
+				$driver = new FileSystem( [ 'path' => APP_ROOT . '/cache' ] );
+			}
+			$cache->setDriver( $driver );
+			return $cache;
+		} );
```

Comment:
This is awesome! No disadvantage to having a local cache, but you could also make it work off the remote Redis via SSH tunnel `ssh -L 4711:enwiki.labsdb:3306 -L 4712:tools-redis:6379 YOU@tools-dev.wmflabs.org` then adjust `.env` accordingly. I should have added that to the README some time ago